src/overlays/actors/ovl_En_Skj/z_en_skj.c

```diff
@@ -39,201 +125,1497 @@ static ColliderCylinderInitType1 sCylinderInit = {
+    } else {
+        func_8002F434(&this->actor, globalCtx, sOcarinaGameRewards[gSaveContext.ocarinaGameReward], 26.0f, 26.0f);
+    }
+}
+
+void EnSkj_FinishOcarinaGameRound(EnSkj* this, GlobalContext* globalCtx) {
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 6) && (func_80106BC8(globalCtx))) {
+        s32 ocarinaGameReward = gSaveContext.ocarinaGameReward;
+
+        if (gSaveContext.ocarinaGameReward < 3) {
+            gSaveContext.ocarinaGameReward++;
+        }
+
+        if (ocarinaGameReward == 2) {
+            gSaveContext.itemGetInf[1] |= 0x80;
+            this->actionFunc = EnSkj_CleanupOcarinaGame;
+        } else {
+            EnSkj_OfferNextRound(this, globalCtx);
+        }
+    }
+}
+
+void EnSkj_CleanupOcarinaGame(EnSkj* this, GlobalContext* globalCtx) {
+    if (sOcarinaMinigameSkullKids[SKULL_KID_LEFT].skullkid != NULL) {
+        sOcarinaMinigameSkullKids[SKULL_KID_LEFT].unk0 = 2;
+    }
+
+    if (sOcarinaMinigameSkullKids[SKULL_KID_RIGHT].skullkid != NULL) {
+        sOcarinaMinigameSkullKids[SKULL_KID_RIGHT].unk0 = 2;
+    }
+
+    if ((sOcarinaMinigameSkullKids[SKULL_KID_LEFT].unk0 == 2) &&
+        (sOcarinaMinigameSkullKids[SKULL_KID_RIGHT].unk0 == 2)) {
+        func_800F5C2C();
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnSkj_OcarinaMinigameShortStumpUpdate(Actor* thisx, GlobalContext* globalCtx) {
+    EnSkj* this = THIS;
+
+    D_80B01EA0 = func_8002F194(&this->actor, globalCtx);
+    this->timer++;
+
+    this->actor.focus.pos.x = 1230.0f;
+    this->actor.focus.pos.y = -90.0f;
+    this->actor.focus.pos.z = 450.0f;
+
+    if (BREG(0) != 0) {
+        DebugDisplay_AddObject(this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z,
+                               this->actor.world.rot.x, this->actor.world.rot.y, this->actor.world.rot.z, 1.0f, 1.0f,
+                               1.0f, 255, 0, 0, 255, 4, globalCtx->state.gfxCtx);
+    }
+
+    this->actionFunc(this, globalCtx);
+
+    this->actor.textId = this->textId;
+    this->actor.xzDistToPlayer = 50.0;
+}
+
+s32 EnSkj_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, void* thisx) {
+    return 0;
+}
+
+void EnSkj_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, void* thisx) {
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_skj.c", 2417);
+
+    if ((limbIndex == 11) && (gSaveContext.itemGetInf[3] & 0x200)) {
+        func_80093D18(globalCtx->state.gfxCtx);
+        Matrix_Push();
+        Matrix_RotateRPY(-0x4000, 0, 0, MTXMODE_APPLY);
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_skj.c", 2430),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, gSKJskullMaskDList);
+        Matrix_Pop();
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_skj.c", 2437);
+}
+
+Gfx* EnSkj_AllocTranslucentDList(GraphicsContext* gfxCtx, u32 alpha) {
```

Comment:
i wouldnt include alloc in the name for these two funcs. it does allocate space for it yea, but its main purpose is to actually provide the display list in general